web/client/components/map/openlayers/DrawSupport.jsx

```diff
@@ -175,6 +175,11 @@ class DrawSupport extends React.Component {
         const olFeature = this.replaceFeatures(newProps);
         if (olFeature) {
             const feature = this.fromOLFeature(olFeature);
+            if (newProps.drawMethod === "Circle" && newProps && newProps.features && newProps.features.length && newProps.features[0] && newProps.features[0].radius >= 0) {
+                // this is needed because the calculation of the radius is wrong
+                // because it is done starting from the approximated polygon
```

Comment:
```suggestion
                // this is because `endDrawing` need to impose the radius value, without any re-calculation or approximation
                // but fromOLFeature re-calculates the radius using some distance formula
                // TODO: externalize radius calculation out from `fromOLFeature`
```